internal/zinc-persist/src/main/scala/sbt/internal/inc/FileBasedStore.scala

```diff
@@ -42,8 +49,10 @@ object FileBasedStore {
             outputStream.closeEntry()
           }
       }
+      IO.move(tmpAnalysisFile, file)
```

Comment:
there is a small risk file can still be corrupted, note `IO.move` does file copy if `renameTo` fails, and file copy may fail

https://github.com/sbt/io/blob/4377f594e1d26d0486792e9ac6dce2c9d2cd9133/io/src/main/scala/sbt/io/IO.scala#L796